frontend/src/app/data-entry/agent-form/agent-form.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Breadcrumb } from '@shared/breadcrumb/breadcrumb.component';
+import { dataIcons } from '@shared/icons';
+import { DataEntryAgentGQL, DataEntryAgentQuery } from 'generated/graphql';
+import { map, Observable, switchMap } from 'rxjs';
+
+@Component({
+  selector: 'lc-agent-form',
+  templateUrl: './agent-form.component.html',
+  styleUrls: ['./agent-form.component.scss']
+})
+export class AgentFormComponent {
+    id$: Observable<string>;
+    data$: Observable<DataEntryAgentQuery>;
+
+    dataIcons = dataIcons;
+
+    constructor(private route: ActivatedRoute, private agentQuery: DataEntryAgentGQL) {
+        this.id$ = this.route.params.pipe(
+            map(params => params['id']),
+        );
+        this.data$ = this.id$.pipe(
+            switchMap(id => this.agentQuery.watch({ id }).valueChanges),
+            map(result => result.data),
+        );
+    }
+
+    getBreadcrumbs(data: DataEntryAgentQuery): Breadcrumb[] {
+        if (data.agentDescription) {
+            return [
+                { link: '/', label: 'Lettercraft' },
```

Comment:
Part of me is a little bit worried that in some circumstances (heavy queries, bad internet connection) the query data may arrive after the DOM has been built and the breadcrumbs won't get updated. What would you say of using an observable (with `async` in the template) or computed signal (with `toSignal(...)`) for this instead to make sure they are fully reactive?

```typescript
breadcrumbs$: Observable<BreadCrumb[]> = this.data$.pipe(
    map(data => ({ ... }))
)
```